crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -46,9 +46,10 @@ def transition(inst, name, state)
 
     @logger.info("Crowbar transition enter: #{name} to #{state}")
 
-    f = acquire_lock "BA-LOCK"
-    begin
-      node = NodeObject.find_node_by_name name
+    pop_it = false
+    node = NodeObject.find_node_by_name name
```

Comment:
Hrm, I just noticed that in the old code, we were fetching the node after acquiring the lock, and now this is different.

I actually believe the old code is more correct here since the lock is there to protect the node (I think?)